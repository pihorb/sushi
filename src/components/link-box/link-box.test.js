import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Router } from 'react-router'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { createMemoryHistory } from 'history'
import LinkBox from './link-box'

const renderWithRouter = (component) => {
  const history = createMemoryHistory()
  return <Router history={history}>{component}</Router>
}

beforeEach(cleanup)

it('check if link exist', () => {
  const { getByTestId } = render(
    renderWithRouter(<LinkBox name='rolls' link='sushi' />)
  )
  const box = getByTestId('link-container')
  const link = getByTestId('link')

  expect(box).toContainElement(link)
})

it('check props rendering', () => {
  const { queryByTestId, getByText, debug } = render(
    renderWithRouter(
      <LinkBox
        name='rolls'
        link='sushi'
        title='rolls'
        subtitle='very good'
        desc='you can eat it every day'
      />
    )
  )

  const title = queryByTestId('link')
  const subtitle = queryByTestId('link-container')
  const description = getByText('you can eat it every day')
  const info = queryByTestId('container-info')

  expect(title).toHaveTextContent('rolls')
  expect(subtitle).toHaveTextContent('very good')
  expect(description).toBeTruthy()
  expect(info).toHaveClass('extra-height')
})

it('snapshot', () => {
  const tree = renderer
    .create(renderWithRouter(<LinkBox link='sushi' />))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
