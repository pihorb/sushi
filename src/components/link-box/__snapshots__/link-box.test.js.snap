// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="link-container"
  data-testid="link-container"
  style={
    Object {
      "background": "rgba(0, 0, 0, 0.5)",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="link-container__info"
    data-testid="container-info"
  >
    <a
      className="link-container__title"
      data-testid="link"
      href="/sushi"
      onClick={[Function]}
    />
  </div>
</div>
`;
